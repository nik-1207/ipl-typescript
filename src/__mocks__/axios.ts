import axios, { AxiosRequestConfig } from "axios";
import { AxiosInterceptorManager } from "axios";

export default {
  get: jest.fn().mockResolvedValue({
    data: {},
  }),
};
