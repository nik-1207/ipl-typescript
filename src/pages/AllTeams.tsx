import { Dispatch, SetStateAction, useEffect, useState } from "react";
import axiosCall from "../api/axios";
import LogoMap from "../config/LogoMap";
import Card from "../components/TeamCard";
import ContainerStyle from "../styles/TeamContainerStyle";

interface teamDataType {
  id: string;
  teamName: string;
  venue: string;
  winningYears: [number];
}
function CardContainer() {
  const teamName: string = window.location.pathname.slice(7);
  const style = ContainerStyle();
  const [TeamData, setTeamData]: [unknown, Dispatch<SetStateAction<unknown>>] =
    useState();
  useEffect(() => {
    axiosCall(setTeamData, teamName);
  }, [teamName]);
  const data = TeamData as [teamDataType];
  return (
    <>
      {data && (
        <div className={style.cardcontainer}>
          {Object.keys(LogoMap).map((key: string, index: number) => {
            return (
              <Card
                img={LogoMap[key]}
                key={index}
                id={data[index].id}
                teamName={data[index].teamName}
                venue={data[index].venue}
                winningYears={data[index].winningYears}
              />
            );
          })}
        </div>
      )}
    </>
  );
}

export default CardContainer;
