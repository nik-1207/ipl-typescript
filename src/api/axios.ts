import axios from "axios";
import { LoadingEvent } from "../event/LoadingEvent";
import { publish } from "../config/PubSub";
import URL from "../config/UrlMap";
export default async function getAllTeamData(callBackData:any, teamName:string)
{
  const url=URL[teamName]
  if(url==='https://ipl-t20.herokuapp.com/teams')
  {

    await axios.get(url).then((res) => {
      callBackData(res.data);
    }).catch(()=>
    {
      callBackData("Blocked Url")
    });
  }else if(url)
  {
    await axios.get(url).then((res) => {
      callBackData(res.data);
    }).catch(()=>
    {
      callBackData({players:["Blocked Url"]})
    })
  }
  else{
    callBackData("invalid  url")
  }

}

axios.interceptors.request.use(
  function (config) {
    console.log("request")
    publish(new LoadingEvent({ isLoading: true }));
    return config;
  },
  function (error) {
    console.log("request Error")
    publish(new LoadingEvent({ isLoading: false }));
    return Promise.reject(error);
  }
);

axios.interceptors.response.use(
  function (response) {
    console.log("response")
    publish(new LoadingEvent({ isLoading: false }));
    return response;
  },
  function (error) {
    console.log("response Error")
    publish(new LoadingEvent({ isLoading: false }));
    return Promise.reject(error);
  }
);