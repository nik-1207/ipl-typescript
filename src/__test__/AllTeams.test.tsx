import { getByTestId, render, waitForElement } from "@testing-library/react";
import axios from "axios";
import { act } from "react-test-renderer";
import AllTeams from "../pages/AllTeams";
const mockedAxios = axios as jest.Mocked<typeof axios>;
it("testing All Teams Page", async () => {
  mockedAxios.get.mockResolvedValueOnce({
    data: [
      {
        id: "chennai-super-kings",
        teamName: "Chennai Super Kings",
        winningYears: [2010, 2011, 2018],
        venue: "M. A. Chidambaram Stadium",
      },
      {
        id: "delhi-capitals",
        teamName: "Delhi Capitals",
        winningYears: [],
        venue: "Feroz Shah Kotla Ground",
      },
      {
        id: "kings-xi-punjab",
        teamName: "Kings XI Punjab",
        winningYears: [],
        venue: "IS Bindra Stadium",
      },
      {
        id: "kolkata-knight-riders",
        teamName: "Kolkata Knight Riders",
        winningYears: [2012, 2014],
        venue: "Eden Gardens",
      },
      {
        id: "mumbai-indians",
        teamName: "Mumbai Indians",
        winningYears: [2013, 2015, 2017, 2019],
        venue: "Wankhede Stadium",
      },
      {
        id: "rajasthan-royals",
        teamName: "Rajasthan Royals",
        winningYears: [2008],
        venue: "Sawai Mansingh Stadium",
      },
      {
        id: "royal-challengers-bangalore",
        teamName: "Royal Challengers Bangalore",
        winningYears: [],
        venue: "M. Chinnaswamy Stadium",
      },
      {
        id: "sunrisers-hyderabad",
        teamName: "Sunrisers Hyderabad",
        winningYears: [2016],
        venue: "Rajiv Gandhi Intl. Cricket Stadium",
      },
    ],
  });
  const { getAllByTestId } = render(<AllTeams />);

  const resolvedSpan = await waitForElement(() => getAllByTestId("cards"));
});
