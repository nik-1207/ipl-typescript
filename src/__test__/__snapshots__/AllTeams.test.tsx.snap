// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing All Teams Page 1`] = `
<div
  class="teamCard-0-2-3 chennai-super-kings-0-2-11"
  data-testid="cards-teams"
>
  <img
    alt="team-logo"
    class="image-0-2-4"
  />
  <div
    class="tab-0-2-10"
  >
    <h2
      class="teamName-0-2-8"
    >
      Chennai Super Kings
    </h2>
    <h3
      class="venue-0-2-9"
    >
      M. A. Chidambaram Stadium
    </h3>
    <div
      class="win-0-2-6"
    >
      <i
        class="fas fa-trophy"
      >
          2010  
          2011  
          2018  
      </i>
    </div>
  </div>
  <p
    class="teamHome-0-2-7"
  >
     Team Home
  </p>
  <i
    class="fas fa-angle-right tab-arrow arrow-0-2-5"
  />
</div>
`;

exports[`testing All Teams Page 2`] = `
<div
  class="teamCard-0-2-3 delhi-capitals-0-2-12"
  data-testid="cards-teams"
>
  <img
    alt="team-logo"
    class="image-0-2-4"
  />
  <div
    class="tab-0-2-10"
  >
    <h2
      class="teamName-0-2-8"
    >
      Delhi Capitals
    </h2>
    <h3
      class="venue-0-2-9"
    >
      Feroz Shah Kotla Ground
    </h3>
  </div>
  <p
    class="teamHome-0-2-7"
  >
     Team Home
  </p>
  <i
    class="fas fa-angle-right tab-arrow arrow-0-2-5"
  />
</div>
`;

exports[`testing All Teams Page 3`] = `
<div
  class="teamCard-0-2-3 kings-xi-punjab-0-2-13"
  data-testid="cards-teams"
>
  <img
    alt="team-logo"
    class="image-0-2-4"
  />
  <div
    class="tab-0-2-10"
  >
    <h2
      class="teamName-0-2-8"
    >
      Kings XI Punjab
    </h2>
    <h3
      class="venue-0-2-9"
    >
      IS Bindra Stadium
    </h3>
  </div>
  <p
    class="teamHome-0-2-7"
  >
     Team Home
  </p>
  <i
    class="fas fa-angle-right tab-arrow arrow-0-2-5"
  />
</div>
`;

exports[`testing All Teams Page 4`] = `
<div
  class="teamCard-0-2-3 kolkata-knight-riders-0-2-14"
  data-testid="cards-teams"
>
  <img
    alt="team-logo"
    class="image-0-2-4"
  />
  <div
    class="tab-0-2-10"
  >
    <h2
      class="teamName-0-2-8"
    >
      Kolkata Knight Riders
    </h2>
    <h3
      class="venue-0-2-9"
    >
      Eden Gardens
    </h3>
    <div
      class="win-0-2-6"
    >
      <i
        class="fas fa-trophy"
      >
          2012  
          2014  
      </i>
    </div>
  </div>
  <p
    class="teamHome-0-2-7"
  >
     Team Home
  </p>
  <i
    class="fas fa-angle-right tab-arrow arrow-0-2-5"
  />
</div>
`;

exports[`testing All Teams Page 5`] = `
<div
  class="teamCard-0-2-3 mumbai-indians-0-2-15"
  data-testid="cards-teams"
>
  <img
    alt="team-logo"
    class="image-0-2-4"
  />
  <div
    class="tab-0-2-10"
  >
    <h2
      class="teamName-0-2-8"
    >
      Mumbai Indians
    </h2>
    <h3
      class="venue-0-2-9"
    >
      Wankhede Stadium
    </h3>
    <div
      class="win-0-2-6"
    >
      <i
        class="fas fa-trophy"
      >
          2013  
          2015  
          2017  
          2019  
      </i>
    </div>
  </div>
  <p
    class="teamHome-0-2-7"
  >
     Team Home
  </p>
  <i
    class="fas fa-angle-right tab-arrow arrow-0-2-5"
  />
</div>
`;

exports[`testing All Teams Page 6`] = `
<div
  class="teamCard-0-2-3 rajasthan-royals-0-2-16"
  data-testid="cards-teams"
>
  <img
    alt="team-logo"
    class="image-0-2-4"
  />
  <div
    class="tab-0-2-10"
  >
    <h2
      class="teamName-0-2-8"
    >
      Rajasthan Royals
    </h2>
    <h3
      class="venue-0-2-9"
    >
      Sawai Mansingh Stadium
    </h3>
    <div
      class="win-0-2-6"
    >
      <i
        class="fas fa-trophy"
      >
          2008  
      </i>
    </div>
  </div>
  <p
    class="teamHome-0-2-7"
  >
     Team Home
  </p>
  <i
    class="fas fa-angle-right tab-arrow arrow-0-2-5"
  />
</div>
`;

exports[`testing All Teams Page 7`] = `
<div
  class="teamCard-0-2-3 royal-challengers-bangalore-0-2-17"
  data-testid="cards-teams"
>
  <img
    alt="team-logo"
    class="image-0-2-4"
  />
  <div
    class="tab-0-2-10"
  >
    <h2
      class="teamName-0-2-8"
    >
      Royal Challengers Bangalore
    </h2>
    <h3
      class="venue-0-2-9"
    >
      M. Chinnaswamy Stadium
    </h3>
  </div>
  <p
    class="teamHome-0-2-7"
  >
     Team Home
  </p>
  <i
    class="fas fa-angle-right tab-arrow arrow-0-2-5"
  />
</div>
`;

exports[`testing All Teams Page 8`] = `
<div
  class="teamCard-0-2-3 sunrisers-hyderabad-0-2-18"
  data-testid="cards-teams"
>
  <img
    alt="team-logo"
    class="image-0-2-4"
  />
  <div
    class="tab-0-2-10"
  >
    <h2
      class="teamName-0-2-8"
    >
      Sunrisers Hyderabad
    </h2>
    <h3
      class="venue-0-2-9"
    >
      Rajiv Gandhi Intl. Cricket Stadium
    </h3>
    <div
      class="win-0-2-6"
    >
      <i
        class="fas fa-trophy"
      >
          2016  
      </i>
    </div>
  </div>
  <p
    class="teamHome-0-2-7"
  >
     Team Home
  </p>
  <i
    class="fas fa-angle-right tab-arrow arrow-0-2-5"
  />
</div>
`;
